test/distributed/partition_kernels.cpp

```diff
@@ -0,0 +1,463 @@
+{
+    using local_index_type = typename TestFixture::local_index_type;
+    using global_index_type = typename TestFixture::global_index_type;
+    const int num_parts = 77;
+    const global_index_type global_size = 13;
+
+    auto part =
+        gko::distributed::Partition<local_index_type, global_index_type>::
+            build_from_global_size_uniform(this->ref, num_parts, global_size);
+    auto dpart =
+        gko::distributed::Partition<local_index_type, global_index_type>::
+            build_from_global_size_uniform(this->exec, num_parts, global_size);
+
+    this->assert_equal(part, dpart);
+}
+
+
+TYPED_TEST(Partition, IsOrderedTrue)
+{
+    using local_index_type = typename TestFixture::local_index_type;
+    using global_index_type = typename TestFixture::global_index_type;
+    comm_index_type num_parts = 7;
+    gko::size_type size_per_part = 1000;
+    gko::size_type global_size = num_parts * size_per_part;
+    gko::Array<comm_index_type> mapping{this->ref, global_size};
+    for (comm_index_type i = 0; i < num_parts; ++i) {
+        std::fill(mapping.get_data() + i * size_per_part,
+                  mapping.get_data() + (i + 1) * size_per_part, i);
+    }
+    auto dpart = gko::distributed::Partition<
+        local_index_type, global_index_type>::build_from_mapping(this->exec,
+                                                                 mapping,
+                                                                 num_parts);
+
+    ASSERT_TRUE(dpart->has_ordered_parts());
+}
+
+
+TYPED_TEST(Partition, IsOrderedFail)
+{
+    using local_index_type = typename TestFixture::local_index_type;
+    using global_index_type = typename TestFixture::global_index_type;
+    comm_index_type num_parts = 7;
+    gko::size_type size_per_part = 1000;
+    gko::size_type global_size = num_parts * size_per_part;
+    gko::Array<comm_index_type> mapping{this->ref, global_size};
+    for (comm_index_type i = 0; i < num_parts; ++i) {
+        std::fill(mapping.get_data() + i * size_per_part,
+                  mapping.get_data() + (i + 1) * size_per_part,
+                  num_parts - 1 - i);
+    }
+    auto dpart = gko::distributed::Partition<
+        local_index_type, global_index_type>::build_from_mapping(this->exec,
+                                                                 mapping,
+                                                                 num_parts);
+
+    ASSERT_FALSE(dpart->has_ordered_parts());
+}
+
+
+TYPED_TEST(Partition, IsOrderedRandom)
+{
+    using local_index_type = typename TestFixture::local_index_type;
+    using global_index_type = typename TestFixture::global_index_type;
+    comm_index_type num_parts = 7;
+    std::uniform_int_distribution<comm_index_type> part_dist{0, num_parts - 1};
+    gko::Array<comm_index_type> mapping{this->ref, 10000};
+    for (gko::size_type i = 0; i < mapping.get_num_elems(); i++) {
+        mapping.get_data()[i] = part_dist(this->rand_engine);
+    }
```

Comment:
can also use generate_random_array

Reply:
thanks, didn't know about that function.